src/cluster/backend/zeromq/ZeroMQ.cc

```diff
@@ -73,6 +73,91 @@ constexpr DebugFlag operator&(uint8_t x, DebugFlag y) { return static_cast<Debug
+                                             return proxy_stats[3];
+                                         });
+    zeek::telemetry_mgr->CounterInstance("zeek", "zeromq_backend_msgs_received", {},
+                                         "Number of messages received by the backend socket", "1", [this]() -> double {
+                                             RefreshStatisticsIfNeeded();
+                                             return proxy_stats[4];
+                                         });
+    zeek::telemetry_mgr->CounterInstance("zeek", "zeromq_backend_bytes_received", {},
+                                         "Number of bytes received by the backend socket", "1", [this]() -> double {
+                                             RefreshStatisticsIfNeeded();
+                                             return proxy_stats[5];
+                                         });
+    zeek::telemetry_mgr->CounterInstance("zeek", "zeromq_backend_msgs_sent", {},
+                                         "Number of messages sent by the backend socket", "1", [this]() -> double {
+                                             RefreshStatisticsIfNeeded();
+                                             return proxy_stats[6];
+                                         });
+    zeek::telemetry_mgr->CounterInstance("zeek", "zeromq_backend_bytes_sent", {},
+                                         "Number of bytes sent by the backend socket", "1", [this]() -> double {
+                                             RefreshStatisticsIfNeeded();
+                                             return proxy_stats[7];
+                                         });
+}
+
+void ZeekProxyTelemetry::RefreshStatisticsIfNeeded() {
+    // Closed? A bit weird, but lets cover this.
+    if ( ! req )
+        return;
+
+    double now = util::current_time();
+    if ( last_updated < now - 0.01 ) {
+        RefreshStatistics();
+        last_updated = util::current_time();
+    }
+}
+
+void ZeekProxyTelemetry::RefreshStatistics() {
+    static std::string cmd = "STATISTICS";
+    static zmq::const_buffer buf{cmd.data(), cmd.size()};
+    zmq::message_t msg;
+    bool more = true;
+
+    // I guess we'll see if this might hang if we query at the wrong time, but
+    // that would lockup the whole node because this code runs on the main thread
+    // and it should be easily recognizable on the stack when attaching via gdb
+    // or sending SIGABRT to dump a core.
+    //
+    // The REQ/REP socket and zmq::proxy_steerable() shouldn't just go away. They
+    // are released by the main thread and this code runs also on the main thread
+    // via the telemetry callbacks, so it should be fine.
+    try {
+        req.send(buf);
+        for ( int i = 0; more; i++ ) {
+            zmq::recv_result_t recv_result = req.recv(msg, zmq::recv_flags::none);
+            proxy_stats[i] = static_cast<double>(*msg.data<uint64_t>());
```

Comment:
Saw that later - better safe than sorry here I guess.

```suggestion
            if ( i < sizeof(proxy_stats) )
                proxy_stats[i] = static_cast<double>(*msg.data<uint64_t>());
            else
                reporter->Warning("Too many ZeroMQ proxy statistics parts - ignoring %d", i);
```